### PR TITLE
Add docker-compose for developing

### LINKED ISSUE #18: Improve experience of first-time development setup
This project should have a clear README which explains how to set up the project, generate documentation, and to create the release build.

Additionally, there should exist a shell- and/or Docker-script which handles the first time setup automatically.

### PR DESCRIPTION
**Pull request by HerrNaN**
Resolves #18

### CODE REVIEW

**Comment by HerrNaN** [comment]
Seems to have a problem when getting http://localhost:8080/pkg/package.js on some machines...

File: frontend/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
-RUN rm /etc/nginx/conf.d/*
-COPY prod.nginx.conf /etc/nginx/conf.d/custom.conf
+# RUN rm /etc/nginx/conf.d/*
+# COPY nginx.conf /etc/nginx/conf.d/custom.conf
 
-COPY index.html /usr/share/nginx/html/index.html
-COPY pkg /usr/share/nginx/html/pkg
-COPY static /usr/share/nginx/html/static
+# Install build dependencies
+# RUN rustup update
```
Comment (hulthe) [inline frontend/Dockerfile]: Remove these comments

File: frontend/dev.nginx.conf [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        listen       80;
+        server_name  localhost;
+
+        #charset koi8-r;
+        #access_log  /var/log/nginx/host.access.log  main;
+
+        # TODO: fix redirection cycle
+        #error_page  404              /404.html;
+        error_page  404              /50x.html;
+        error_page  500 502 503 504  /50x.html;
```
Comment (hulthe) [inline frontend/dev.nginx.conf]: I think we can remove this.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      - '3000:3000'
+    depends_on:
+      - backend
+
+  backend:
+    build:
+      context: .
+      dockerfile: ./backend/Dockerfile
+    ports:
+      - '8000:8000'
```
Comment (hulthe) [inline docker-compose.yml]: We don't need to expose the ports of the individual services, they are accessed through the router.

**Review by hulthe — CHANGES_REQUESTED** [review]
Some small cleanup left to do. And we still need to test this.

**Comment by hulthe** [comment]
It works! At least on my machine...

The only problem with this is that it produces build artifacts that are owned by `root`...
I think it would be best to have the docker container inherit the UID & GID of the user.

**Comment by hulthe** [comment]
Another solution might be to mount `target` in its own anonymous volume, hm...

**Comment by hulthe** [comment]
Closing due to staleness

**Pull request closed without merging**